### ISSUE TITLE
Integrate with Running sidebar

### ISSUE THREAD

**Issue by bollwyvl** [jupyter-lsp/jupyterlab-lsp#205]
## Elevator Pitch

Add language server status to the JupyterLab _Running_ sidebar.

## Motivation

_Running_ is the place where users will expect to be able to get information about processes running on their behalf.

## Design Ideas

- make `ILanguageServerManager`, `IConnectionManager` (or some other wrapper around it, but whatever) extend `IRunningSessions.IManager`

```ts
    // Name that is shown to the user
    name: string;
    // called when the shutdown all button is pressed
    shutdownAll(): void;
    // list the running models.
    running(): IRunningItem[];
    // Force a refresh of the running models.
    refreshRunning(): void;
    // A signal that should be emitted when the item list has changed.
    runningChanged: ISignal<any, any>;
```
- make something (probably an `ILanguageServer`) that extends `IRunningSessions.IRunningItem`

```
    // called when the running item is clicked
    open: () => void;
    // called when the shutdown button is pressed on a particular item
    shutdown: () => void;
    // LabIcon to use as the icon
    icon: () => LabIcon;
    // called to determine the label for each item
    label: () => string;
    // called to determine the `title` attribute for each item, which is revealed on hover
    labelTitle?: () => string;
```
- wire up the server status report from the backend (and refresh it occasionally, like when some new websocket occurs)
- add a `DELETE` handler for each server to actually trigger the shutdown
- handle killing the websocket and the server in the correct order
- decide on an icon approach
  - we could muddy things a bit, and drive this off the status of the websocket, a la the status bar
  - or pull the file icon for its first language/extension
  - or add an svg icon to the server spec to make it prettier... but a lot of the servers don't have branding, and the language is 
- decide on a label approach, probably `display_name`, but we could get fancy
- decide what to do when you click
  - initially, it won't do much of anything, and shouldn't block this
  - once #196 lands, redirect to whatever that is, as it's something about the actual server (rather than documents, etc) and could have links to docs, issues, etc. for that server, show the response time information, etc.

**Comment by bollwyvl** [comment]
hooray, 1.0.0: this issue wasn't going to get done today (or this weekend). Not sure what release planning looks like, but since we've gone ahead and released, this will almost certainly require a `1.1`. Since we have no public API on the frontend, it won't be more than that, but the `ws` package might see significant changes again.

I can pretty much get it to work by killing the reader/writer/process, but the frontend gets _very_ confused. If i take out the "ignore" stuff, and delete the connection, re-opening the document works fine, but it definitely can't re-connect to an open document.

I don't think we can rely on the LSP `shutdown` message, because where we really need the hammer is for badly-behaved processes.

This is pointing to the need to deal with the lifecycle of the language server session more explicity, probably at a level above the connection. much like lab 2.0 does for kernels: this might mean a frontend `Session` that listens for `close` and proxies through the connection API, and knows how to re-establish connections, etc.

**Comment by krassowski** [comment]
This now works in JupyterLab 4 with jupyterlab-lsp 5.0.
